hack/lib/golang.sh

```diff
@@ -24,6 +24,7 @@ readonly KUBE_GOPATH="${KUBE_OUTPUT}/go"
 # If you update this list, please also update build/release-tars/BUILD.
```

Comment:
To be completely honest, I read the comment immediately before it that said:

    # Note: if you are adding something here, you might need to add it to
    # kube::build::source_targets in build/common.sh as well.

and must have stopped reading because `source_targets` doesn't exist in `build/common.sh` (or anywhere else in the codebase, says `grep -r`). :-)

Given that we don't actually want this in the tar, and I just want to ensure that the binary gets built when users run `make` (or `make bazel-build`), I'll just leave this alone and move the target to `KUBE_ALL_TARGETS`.